libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -62,7 +61,13 @@ impl ConnectionManager {
 
     pub fn start(&mut self) -> Result<Connector, ConnectionManagerError> {
         let (sender, recv) = sync_channel(CHANNEL_CAPACITY);
-        let mut state = self.connection_state.clone();
+        let mut state = if let Some(state) = self.connection_state.take() {
```

Comment:
This `if let ... else ... ` could be simplified by using `self.connection_state.take().ok_or_else(|| Err(...))?`